cirq-google/cirq_google/ops/internal_gate.py

```diff
@@ -78,3 +99,46 @@ def _value_equality_values_(self):
             self._num_qubits,
             frozenset(self.gate_args.items()) if hashable else self.gate_args,
         )
+
+
+def encode_function(
+    x: Union[Sequence[float], np.ndarray],
+    y: Sequence[float],
+    msg: Optional[program_pb2.CustomArg] = None,
+) -> program_pb2.CustomArg:
+    """Encodes a general function as a list of evaluations.
+
+    This method discretizes a function into a list of point evaluations. Evaluating the
+    function on general points is then done by interpolating the function using the given
+    method.
+
+    Args:
+        x: Sequence of values of the free variable.
+            For 1D functions, this input is assumed to be given in increasing order.
+        y: Sequence of values of the dependent variable.
+            Where y[i] = func(x[i]) where `func` is the function being encoded.
+        msg: Optional CustomArg to serialize to.
+            If not provided a CustomArg is created.
+
+    Returns:
+        A CustomArg encoding the function.
+
+    Raises:
+        ValueError: If
+            - `x` is 1D and not sorted in increasing order.
+            - `method` is not supported.
```

Comment:
Can we add a check if `x` and `y` have matching shapes?

```suggestion
            - `x` is not sorted or has 2 or more dimensions.
            - `y` has a different shape than `x`
```

Reply:
fixed docstring.